### ISSUE TITLE
(🎁) new rule: redundant comprehension

### ISSUE THREAD

**Issue by KotlinIsland** [PyCQA/pylint#7843]
### Current problem

I am seeing suff in the wild like:
```py
";".join([f"{k}:{v}" for k, v in something])
```

The list comprehension here is redundant and you should really be writing:
```py
";".join(f"{k}:{v}" for k, v in something)
```

### Desired solution

Detect these redundant calls and report against them.

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue. This feel like a false negative for ``use-a-generator`` or ``consider-using-a-generator``.

**Comment by clavedeluna** [comment]
Could author / maintainer update the title?
